src/lib/form-field/form-field.scss

```diff
@@ -144,6 +144,8 @@ $mat-form-field-default-infix-width: 180px !default;
   width: 100%;
   // Need this so that the underline doesn't block the hover effect.
   pointer-events: none;
+  // We transform the height slightly higher to fix inconsistent underline height for some DPIs.
+  transform: scaleY(1.0001);
```

Comment:
This feels like one of those things where it only really helps at certain screen sizes and/or DPIs. I don't think it would hurt having it, though.

Reply:
I tested it out at different zoom levels on my desktop and retina macbook and this definitely helped make things more consistent and made the underlines not disappear on zoom levels between 50% and 100%